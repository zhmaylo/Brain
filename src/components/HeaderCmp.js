import React from 'react';
import { View, StyleSheet, TextInput, TouchableOpacity, Image } from 'react-native';
import { WINDOW_WIDTH, WINDOW_HEIGHT } from './../constants/otherConst';
import { TEXT_INPUT_PLACEHOLDER, ICON_HEADER, HEADER_BUTTON_SIZE } from './../constants/headerConst';

import { MenuSortCmp} from './MenuSortCmp';


export const HeaderCmp = (productsList, menuUrl, dispatch) => {

    return (
        <View style={styles.container}>
            <TouchableOpacity
               onPress={() => { menuUrl() }}
            >
                <Image
                    style={styles.menuImage}
                    // style={{ width: 26, height: 26, marginRight: 20 }}
                    source={ICON_HEADER [0].iconSource}
                />
            </TouchableOpacity>

            {/* <Image
                style={styles.imageFind}
                // style={{ backgroundColor: "green" }}
                source={require('./../icons/Search-icon.png')}
            >
            </Image> */}
            <TextInput
                style={styles.search}
                placeholder= {TEXT_INPUT_PLACEHOLDER}
                >
            </TextInput>
            <TouchableOpacity 
                onPress={() => {
                                MenuSortCmp ();
                                console.log(' onPress=>MenuSortCmp called');
                                // <MenuSortCmp />
                                // let prodList = buttonSort(productsList) 
                                
                                // dispatch({ type: 'PRODUCTS_LIST', payload: prodList })
                                }
                        }
            >
                <Image
                    style={styles.menuImage}
                    source={ICON_HEADER [1].iconSource}
                >
                </Image>
            </TouchableOpacity>
            <TouchableOpacity
                onPress={() => { menuUrl() }}
            >
                <Image
                    style={styles.menuImage}
                    source={ICON_HEADER [2].iconSource}
                >
                </Image>
            </TouchableOpacity>
            
        </View >
    )
}


const styles = StyleSheet.create({
    menuImage: {
        width: HEADER_BUTTON_SIZE,
        height: HEADER_BUTTON_SIZE,
    },

    search: {
        width: WINDOW_WIDTH-3*HEADER_BUTTON_SIZE-20,
        height: HEADER_BUTTON_SIZE,
        borderWidth: 2,
        paddingLeft: 4,
        marginHorizontal: 5,

    },
    container: {
        flex: 1,
        flexDirection: "row",
        justifyContent: "space-between",
        alignItems: "flex-start",
        width: WINDOW_WIDTH,
        paddingVertical: 3,
        paddingHorizontal: 3,
        opacity: 0.8,
    }
})
