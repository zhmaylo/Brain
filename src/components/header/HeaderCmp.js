import React from '../../../node_modules/react';
import { View, StyleSheet, TextInput, TouchableOpacity, Image } from 'react-native';
import { WINDOW_WIDTH } from '../../constants/otherConst';
import { TEXT_INPUT_PLACEHOLDER, ICON_HEADER, HEADER_BUTTON_SIZE } from '../../constants/headerConst';

import { CATEG_SCR } from '../../constants/appNavigatorConst';
import { SORT_SCR, FILTER_SCR } from './../../constants/appNavigatorConst';
import SearchScr from '../../screen/SearchScr';

export const HeaderCmp = (props) => {

    return (
        <View style={styles.container}>
            <TouchableOpacity
                onPress={() => { props.navigation.navigate(CATEG_SCR) }}
  
            >
                <Image
                    style={styles.menuImage}
                    source={ICON_HEADER[0].iconSource}
                />
            </TouchableOpacity>

            {/* <TextInput
                style={styles.search}
                placeholder={TEXT_INPUT_PLACEHOLDER}
            >
            </TextInput> */}

            <SearchScr/>

            <TouchableOpacity className="button_sort"
                onPress={() => {
                    props.navigation.navigate(SORT_SCR);
                }}
            >

                <Image
                    style={styles.menuImage}
                    source={ICON_HEADER[1].iconSource}
                >
                </Image>
            </TouchableOpacity>


            <TouchableOpacity
                onPress={() => {
                    props.navigation.navigate(FILTER_SCR);
                }}
            >
                <Image
                    style={styles.menuImage}
                    source={ICON_HEADER[2].iconSource}
                >
                </Image>
            </TouchableOpacity>

        </View >
    )
}


const styles = StyleSheet.create({
    menuImage: {
        width: HEADER_BUTTON_SIZE,
        height: HEADER_BUTTON_SIZE,
    },


    container: {
        flex: 1,
        flexDirection: "row",
        justifyContent: "space-between",
        alignItems: "flex-start",
        width: WINDOW_WIDTH,
        paddingVertical: 3,
        paddingHorizontal: 3,
        opacity: 0.8,
    }
})
