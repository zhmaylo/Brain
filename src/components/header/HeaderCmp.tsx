import React from 'react';
import { View, StyleSheet, TouchableOpacity, Image } from 'react-native';
import { WINDOW_WIDTH } from '../../constants/otherConst';
import { ICON_HEADER, HEADER_BUTTON_SIZE } from '../../constants/headerConst';

import { CATEG_SCR } from '../../constants/appNavigatorConst';
import { SORT_SCR, FILTER_SCR } from './../../constants/appNavigatorConst';
import { SearchCmp } from '../search/SearchCmp';

export const HeaderCmp = ({ props, dispatch }) => {

    return (
        <View style={styles.container}>
            <TouchableOpacity
                onPress={() => { props.navigation.navigate(CATEG_SCR) }}
            >
                <Image
                    style={styles.menuImage}
                    source={ICON_HEADER[0].iconSource}
                />
            </TouchableOpacity>
            <View style={styles.searchCmp} >
                <SearchCmp
                    onChangeRequest={(type: any, payload: any) => {
                        dispatch({ type: type, payload: payload });
                    }}
                />
            </View>
            <TouchableOpacity
                onPress={() => {
                    props.navigation.navigate(SORT_SCR);
                }}
            >
                <Image
                    style={styles.menuImage}
                    source={ICON_HEADER[1].iconSource}
                >
                </Image>
            </TouchableOpacity>
            <TouchableOpacity
                onPress={() => {
                    props.navigation.navigate(FILTER_SCR);
                }}
            >
                <Image
                    style={styles.menuImage}
                    source={ICON_HEADER[2].iconSource}
                >
                </Image>
            </TouchableOpacity>
        </View >
    )
}


const styles = StyleSheet.create({
    menuImage: {
        width: HEADER_BUTTON_SIZE,
        height: HEADER_BUTTON_SIZE,
    },

    container: {
        flex: 1,
        flexDirection: "row",
        justifyContent: "space-between",
        alignItems: "flex-start",
        width: WINDOW_WIDTH,
        // paddingVertical: 3,
        paddingTop: 3,
        paddingHorizontal: 3,
        opacity: 0.8,
    },
    searchCmp: {
        width: WINDOW_WIDTH - HEADER_BUTTON_SIZE * 4,
    }
})
