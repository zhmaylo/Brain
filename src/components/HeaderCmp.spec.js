
import React from 'react';
import renderer from 'react-test-renderer';
import { HeaderCmp } from './HeaderCmp';


const callback = () => {
    console.log("Test headerCmp - callback test  ");
    // expect(data).toMatch('Test Ok');
    return;
};

test('renders correctly', () => {
    
    const tree = renderer.create(
        HeaderCmp(callback)           )
        .toJSON();
    expect(tree).toMatchSnapshot();
});




// tasca - testAsyncCallback Creates Async Callback test
// taspr - testAsyncPromise Creates Aysnc Promise test
// tasaw - testAsyncAwait Creates Async Await test
// tsnsh - testSnapShot Creates Snapshot test
// tshre - testShallowRenderer Creates Shallow Renderer test