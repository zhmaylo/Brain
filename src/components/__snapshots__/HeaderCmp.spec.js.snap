// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "flex-start",
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "opacity": 0.8,
      "paddingHorizontal": 3,
      "paddingVertical": 3,
      "width": 750,
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/icons/menu.png",
        }
      }
      style={
        Object {
          "height": 30,
          "width": 30,
        }
      }
    />
  </View>
  <TextInput
    allowFontScaling={true}
    placeholder="Поиск"
    rejectResponderTermination={true}
    style={
      Object {
        "borderWidth": 2,
        "height": 30,
        "marginHorizontal": 5,
        "paddingLeft": 4,
        "width": 670,
      }
    }
    underlineColorAndroid="transparent"
  />
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/icons/filter.png",
        }
      }
      style={
        Object {
          "height": 30,
          "width": 30,
        }
      }
    />
  </View>
</View>
`;
